app/models/range_progresses.rb

```diff
@@ -0,0 +1,68 @@
+class RangeProgresses
+  def initialize(id, start_date, end_date)
+    @start_date = start_date
+    @end_date = end_date
+    @progresses = DailyProgress.where(practice_id: id, done_at: start_date..end_date)
+    @done_dates = @progresses.where('progress > 0').pluck(:done_at).sort
+  end
+
+  def format_each_week
+    (@start_date..@end_date).each_with_object([]) do |date, formated_dates|
+      if formated_dates.blank? || formated_dates[-1].last.wday == 6
```

Comment:
__[reek]__
* [DuplicateMethodCall] calls formated_dates[(-1)] 2 times
* [FeatureEnvy] refers to formated_dates more than self
* [FeatureEnvy] refers to formated_dates more than self